api/envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -50,10 +50,22 @@ message Endpoint {
   HealthCheckConfig health_check_config = 2;
 }
 
+// Named reference to an Endpoint.
+message NamedEndpoint {
+  // Identifier for this endpoint. Should be unique within a given ClusterLoadAssignment.
+  string name = 1 [(validate.rules).string.min_bytes = 1];
+
+  // Upstream host idenfier.
+  endpoint.Endpoint endpoint = 2;
+}
+
 // An Endpoint that Envoy can route traffic to.
 message LbEndpoint {
-  // Upstream host identifier
-  Endpoint endpoint = 1;
+  // Upstream host identifier or a named reference.
+  oneof host_identifier {
+    Endpoint endpoint = 1;
+    NamedEndpoint named_endpoint = 5;
```

Comment:
i think what you have makes the most sense, but what's the tradeoff here over adding the optional identifier to the `Endpoint` message itself?

Reply:
Yeah, I reckon conceptually it might be simpler to just add an optional named field to the endpoint directly; the anonymous endpoints that we have today then fit neatly into this with proto defaults.